scripts/us_census/acs5yr/subject_tables/common/generate_col_map.py

```diff
@@ -0,0 +1,172 @@
+)
+from statvar_dcid_generator import get_stat_var_dcid
+
+
+def generate_stat_var_map(spec_dict: dict, column_list: list) -> dict:
+    """Wrapper function for generateColMapBase class to generate column map.
+
+  Args:
+    specDict:
+      A dictionary containing specifications for the different properties of the statistical variable.
+    columnList:
+      A list of column names for which the column map needs to be generated. This is typically the column header in the dataset.
+
+  Returns:
+    A dict mapping each column to their respective stat_var node definitions.
+    Example output: {
+      "Total Civilian population": {
+        "populationType": "Person",
+        "statType": "measuredValue",
+        "measuredProperty": "Count Person"
+        "armedForceStatus": "Civilian"
+      },
+      "<column-name-2>": {}, .....
+    }
+  """
+    col_map_obj = GenerateColMapBase(spec_dict=spec_dict,
+                                     column_list=column_list)
+    return col_map_obj._generate_stat_vars_from_spec()
+
+
+class GenerateColMapBase:
+    """module to generate a column map given a list of columns of the dataset and a JSON Spec
+
+  Attributes:
+    specDict: A dictionary containing specifications for the different properties of the statistical variable.
+    columnList: A list of column names for which the column map needs to be generated. This is typically the column header in the dataset.
+    delimiter: The delimiting string that is used for tokenising the column name
+  """
+
+    def __init__(self, spec_dict={}, column_list=[], delimiter='!!'):
+        """module init"""
+        self.features = spec_dict
+        self.column_list = column_list
+        self.column_map = {}
+        self.delimiter = delimiter
+
+    def _update_stat_var_dict(self, stat_var, part_list):
+        """changes certain properties of the stat_var dict based on column name"""
+        # if Margin of Error columns are present change the statType
+        if 'Margin of Error' in part_list:
+            stat_var['statType'] = 'marginOfError'
+        return stat_var
+
+    def _generate_stat_vars_from_spec(self):
+        """generates stat_var nodes for each column in column list and returns a dictionary of columns with their respective stat_var nodes
+        Example output: {
+      "Total Civilian population": {
+        "populationType": "Person",
+        "statType": "measuredValue",
+        "measuredProperty": "Count Person"
+        "armedForceStatus": "Civilian"
+      },
+      "<column-name-2>": {}, .....
+    }"""
+        # for each column generate the definition of their respective statistical variable node
+        for col in self.column_list:
+            if 'ignoreColumns' in self.features and col not in self.features[
+                    'ignoreColumns']:
+                self.column_map[col] = self._column_to_statvar(col)
```

Comment:
Should we generate an error _column_to_statvar() returns an empty statvar?

Reply:
I have marked this as a [TODO](https://github.com/sharadshriram/data/blob/acs_common/scripts/us_census/acs5yr/subject_tables/common/generate_col_map.py#L209) in the code.